src/mapreduce.jl

```diff
@@ -160,25 +160,76 @@ end
 ## reducedim ##
 ###############
 
-@inline reducedim(op, a::StaticArray, ::Val{D}) where {D} = mapreducedim(identity, op, a, Val{D})
-@inline reducedim(op, a::StaticArray, ::Val{D}, v0) where {D} = mapreducedim(identity, op, a, Val{D}, v0)
+@inline reducedim(op, a::StaticArray, ::Type{Val{D}}) where {D} = mapreducedim(identity, op, a, Val{D})
```

Comment:
When making this change, I would like to have things like 
```julia
default_similar_type{T,S,D}(::Type{T}, s::Size{S}, ::Type{Val{D}})
```
in `abstractarray.jl` change to
```julia
default_similar_type{T,S,D}(::Type{T}, s::Size{S}, ::Val{D})
```
as well.  Because such a change is outside `mapreduce.jl`, I think creating a separate PR dedicated to `::Type{Val{D}}` -> `::Val{D}` would be cleaner...  So yes, I would prefer not to make this change at the moment.